agent/dns.go

```diff
@@ -212,7 +213,29 @@ func (cfg *dnsConfig) GetTTLForService(service string) (time.Duration, bool) {
 	return 0, false
 }
 
-func (d *DNSServer) ListenAndServe(network, addr string, notif func()) error {
+// TODO: rename to newDNSServer once DNSServer has been renamed to DNSHandler
+func newAPIServerDNS(handler *DNSServer, addr net.Addr, notify func()) apiServer {
+	handler.setupMux()
+
+	srv := &dns.Server{
+		Addr:              addr.String(),
+		Net:               addr.Network(),
+		Handler:           handler.mux,
+		NotifyStartedFunc: notify,
+	}
+	if addr.Network() == "udp" {
+		srv.UDPSize = 65535
+	}
+
+	return apiServer{
+		Protocol: "dns",
+		Addr:     addr,
+		Run:      srv.ListenAndServe,
```

Comment:
I think this will need to log the error from `ListenAndServe`, otherwise we won't see it until `Shutdown`.